### ISSUE TITLE
"goto module" not working for deep link into node_module

### ISSUE THREAD

**Issue by ljharb** [Galooshi/import-js#274]
I have a line `import withGlobal from 'mocha-wrap/withGlobal';`. This is `withGlobal.js` inside `node_modules/mocha-wrap`.

However, in 0.7.4 in sublime, when I "goto module", it takes me to an empty file in "$PROJECT_ROOT/mocha-wrap/withGlobal". I suspect the fix to #273 may have broken this use case?

**Comment by ljharb** [comment]
Specifically, the project root should be used exclusively for relative requires (starting with `.`), the filesystem root for requires starting with `/`, and the `node_modules` resolution algorithm for any other requires. `require.resolve()` should be able to handle all of those - if you chdir to the project root and run `require.resolve` on all paths, I'd expect that to work.

**Comment by trotzig** [comment]
Thanks for the report @ljharb! This wasn't introduced by my fix for #273. It's been present for a while. I think we've missed it because I'm rarely importing files inside package dependencies. 

Thanks for the pointer to `require.resolve`. It allowed me to remove some custom (broken) resolution code.

**Comment by trotzig** [comment]
@ljharb: since @lencioni published a new version, all you have to do is to `npm install -g import-js`. No need to update the Sublime plugin.

**Comment by ljharb** [comment]
@trotzig hmm, now it opens the right file but it opens empty, even though the file has code in it.

**Comment by trotzig** [comment]
Interesting. Do you have that module `npm link`ed by any chance? I'm able to reproduce it with such a dependency. 

I found that out by making `chokidar` `npm link`ed.
```
> cd ~/chokidar
> npm link
> cd ~/my-project
> npm link chokidar
> importjs goto chokidar some-file.js
{ [...] "goto": "/Users/henrictrotzig/import-js/node_modules/chokidar/index.js" }
```

The resulting `goto` in the json output is exactly what `require.resolve` tells us. I'm surprised to see `import-js` being part of the resulting (absolute) path. That doesn't happen for dependencies that aren't `npm link`ed (in which case I get a relative path, e.g. `node_modules/chokidar`).

Since `importjs` is also `npm link`ed for me locally (which might interact weirdly here), it would be interesting to see what the resulting `goto` path is for you. What does Sublime tell you if you hover over the opened file's tab?

**Comment by ljharb** [comment]
In this case neither are `npm link`ed, although I'd expect that to work too. When I hover over the tab, the path is correct (and when I open that path on the command line, it contains the right code).

**Comment by trotzig** [comment]
I filed a separate bug for the issue with linked dependencies: https://github.com/Galooshi/import-js/issues/279

I was able to reproduce the open file issue now, and I pushed a fix in daefb04318b4e. Thanks for your patience.
